core/reactor_common.c

```diff
@@ -452,7 +471,11 @@ void _lf_start_time_step() {
                     && _lf_tokens_with_ref_count[i].status != NULL) {
                 *(_lf_tokens_with_ref_count[i].status) = absent;
             }
-            _lf_done_using(*(_lf_tokens_with_ref_count[i].token));
+            // If the reference count is already zero, no need to decrement.
+            if (*(_lf_tokens_with_ref_count[i].token)
+                    && (*(_lf_tokens_with_ref_count[i].token))->ref_count > 0) {
+                _lf_done_using(*(_lf_tokens_with_ref_count[i].token));
+            }
```

Comment:
What is going on here? At the beginning of a time step we look through all tokens which is referenced somewhere. Those could be tokens statically connected to actions or ports. If reset_is_present=True then reset. If token=NULL and the reference count is greater than zero, then call `_lf_done_using`. So essentially all tokens that are present (or was present last time step?) have their reference count decremented? I don't follow that logic, is this cleanup after previous time tag?